src/repo.rs

```diff
@@ -1054,6 +1054,14 @@ impl Repository {
         }
     }
 
+    /// Override the object database for this repository
+    pub fn set_odb(&self, odb: &Odb<'_>) -> Result<(), Error> {
+        unsafe {
+            try_call!(raw::git_repository_set_odb(self.raw(), odb.raw()));
```

Comment:
Can you double check the memory management here? Is it safe to drop `Odb` after this call?

Reply:
So here, yes, it seems safe to drop `Odb` after this call. libgit2 internally does refcounting on the odb objects, and will only free the odb once it is not "owned" and the refcount drops to zero. Calling `git_repository_set_odb` will both mark the repo as the owner and increment the refcount, so if the rust code drops the ref it will continue to live until the repo itself gets destroyed or has a different odb installed.